### PR TITLE
test(css-modules): add integration test validating di

### PR DESCRIPTION
**Pull request by EisenbergEffect**
# Pull Request

## 📖 Description

This PR adds a simple integration test that validates that two nested custom element's views have the correct class lists internally. The parent view contains css modules behavior, while the child view doesn't. Both views use the same class name, but only the parent view gets its class mapped via CSS modules.

### 🎫 Issues

This is a follow-up PR related to the recent DI hierarchy fixup, providing an additional use case related to expected CSS modules behavior.

### CODE REVIEW

File: packages/__tests__/jit-html/styles.integration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        name: 'with-styles',
+        template: `<div id="target" class="test"><slot></slot></div>`,
+        dependencies: [styles(cssClasses)]
+      },
+      class {
+        static get inject() { return [INode]; }
+        constructor(private readonly element: HTMLElement) {}
+
+        public attached() {
+          // verify that the div has an updated class
```
Comment (fkleuver) [inline packages/__tests__/jit-html/styles.integration.spec.ts]: @EisenbergEffect Is this still a TODO from your end or was this something you wanted me to add?

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30622-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by EisenbergEffect** [comment]
@fkleuver This was actually ready to go when I first submitted it, yes.

File: packages/__tests__/jit-html/styles.integration.spec.ts (hunk 1 above)
Comment (EisenbergEffect) [reply packages/__tests__/jit-html/styles.integration.spec.ts]: Sorry, that is not a valid comment. Oops. I forgot to remove that.

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30780-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://31886-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Pull request merged**